### PR TITLE
Added http proxy options to configuration

### PR DESCRIPTION
**Pull request by baloo42**
added http and https proxy settings.

Tried to add options for _nonProxyHosts_ settings, but I didn't get it to work. Maybe there is sombody who can help?

```java
  /**
   * Set hosts that should not be proxied. Accepts Java "http.nonProxyHosts" wildcard patterns (one per line, no '|'
   * hostname delimiters).
   * Previously configured values can be removed by calling this with no parameters.
   */
  void nonProxyHosts(String... nonProxyHosts);
```
From:
https://github.com/sonatype/nexus-public/blob/tag-3.3.1-01/components/nexus-core/src/main/java/org/sonatype/nexus/CoreApi.java

### CODE REVIEW

**Comment by dannyk81** [comment]
I was able to get `nonProxyHosts` to work, however I think there's something wrong with this API.

based on the API doc, I assume that `core.nonProxyHosts` should be called for each item, however they just overwrite each other and I end up with just the last entry - is that what you've experienced?

I added this to the end of your groovy script:
```
if (parsed_args.with_http_proxy) {
    core.nonProxyHosts()
    parsed_args.non_proxy_hosts.each { core.nonProxyHosts(it) }
}
```

defaults:
```
nexus_non_proxy_hosts:
  - "localhost"
  - "127.*"
  - "[::1]"
```

tasks:
```
      non_proxy_hosts: "{{ nexus_non_proxy_hosts }}"
```

**Comment by samherve** [comment]
There seem to be a trick with groovy calling java varargs : have you tried with something like `core.nonProxyHosts(parsed_args.proxy_exclude_hosts as String[])` ?

File: tasks/main.yml [hunk 1]
```diff
@@ -145,6 +145,21 @@
...
+  vars:
+    script_name: setup_http_proxy
+    args:
+      with_http_proxy: "{{ nexus_with_http_proxy }}"
+      http_proxy_host: "{{ nexus_http_proxy_host }}"
+      http_proxy_port: "{{ nexus_http_proxy_port }}"
+      http_proxy_username: "{{ nexus_http_proxy_username }}"
+      http_proxy_password: "{{ nexus_http_proxy_password }}"
+      with_https_proxy: "{{ nexus_with_https_proxy }}"
+      https_proxy_host: "{{ nexus_http_proxy_host }}"
```
Comment (samherve) [inline tasks/main.yml]: There seem to be a typo here ? (missing "s"). Same on next line.

**Review by samherve — CHANGES_REQUESTED** [review]

**Comment by dannyk81** [comment]
@samherve That did it!

with `core.nonProxyHosts(parsed_args.proxy_exclude_hosts as String[])` it works perfectly!

**Comment by baloo42** [comment]
thanks!

fixed typo and added new option nexus_proxy_exclude_hosts

**Pull request merged**